libs/core/src/service/activity_service.rs

```diff
@@ -0,0 +1,25 @@
+use std::collections::HashMap;
+
+use crate::Requester;
+use crate::{CoreResult, RequestContext};
+use itertools::Itertools;
+use uuid::Uuid;
+
+impl<Client: Requester> RequestContext<'_, '_, Client> {
+    pub fn most_active_documents(&mut self) -> CoreResult<Vec<(Uuid, usize)>> {
+        let mut result: HashMap<Uuid, usize> = HashMap::new();
+
+        for (k, v) in self.tx.read_activity.get_all() {
+            result.insert(*k, if result.contains_key(k) { *v } else { *v + result[k] });
```

Comment:
If `result` does contain a key already, wouldn't you add what it has? (in other words, I think you should take the inverse of your `if` statement)

Reply:
To get rid of the if statement, you can use this: `result.get(k).unwrap_or_else(0) + *v`.

Reply:
ty for the oneliner!